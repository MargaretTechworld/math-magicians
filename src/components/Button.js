import React from 'react';
import PropTypes from 'prop-types';

function Button({ value, className }) {
  return <button type="button" className={className}>{value}</button>;
}

export default Button;

Button.propTypes = {
  value: PropTypes.string.isRequired,
  className: PropTypes.string.isRequired,
};
