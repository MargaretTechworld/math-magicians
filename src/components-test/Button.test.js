import React from 'react';
import renderer from 'react-test-renderer';
import Button from '../components/Button';

test('renders correctly', () => {
  const handleClick = () => <></>;

  const tree = renderer
    .create(<Button value="+" className="button-class" onClick={handleClick} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
