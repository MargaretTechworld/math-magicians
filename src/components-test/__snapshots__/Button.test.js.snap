// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="button-class"
  onClick={[Function]}
  type="button"
>
  +
</button>
`;
